src/main/java/com/clubobsidian/dynamicgui/listener/InventoryInteractListener.java

```diff
@@ -0,0 +1,188 @@
+/*
+   Copyright 2020 Club Obsidian and contributors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package com.clubobsidian.dynamicgui.listener;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map.Entry;
+
+import com.clubobsidian.dynamicgui.entity.PlayerWrapper;
+import com.clubobsidian.dynamicgui.event.inventory.Click;
+import com.clubobsidian.dynamicgui.event.inventory.InventoryClickEvent;
+import com.clubobsidian.dynamicgui.event.inventory.InventoryDragEvent;
+import com.clubobsidian.dynamicgui.gui.Gui;
+import com.clubobsidian.dynamicgui.gui.InventoryView;
+import com.clubobsidian.dynamicgui.gui.Slot;
+import com.clubobsidian.dynamicgui.inventory.InventoryWrapper;
+import com.clubobsidian.dynamicgui.inventory.ItemStackWrapper;
+import com.clubobsidian.dynamicgui.manager.dynamicgui.GuiManager;
+import com.clubobsidian.dynamicgui.parser.function.FunctionType;
+import com.clubobsidian.dynamicgui.parser.function.tree.FunctionNode;
+import com.clubobsidian.dynamicgui.util.FunctionUtil;
+
+import com.clubobsidian.trident.EventHandler;
+
+public class InventoryInteractListener {
+
+	@EventHandler
+	public void invClick(final InventoryClickEvent e)
+	{
+		PlayerWrapper<?> player = e.getPlayerWrapper();
+		if(!GuiManager.get().hasGuiOpen(player))
+		{
+			return;
+		}
+		
+		Gui gui = GuiManager.get().getCurrentGui(player);
+		
+		Slot slot = this.getSlotFromIndex(gui, e.getSlot());
+		if(slot == null)
+		{
+			e.setCanceled(true);
+			return;
+		}
+		
+		ItemStackWrapper<?> item = e.getItemStackWrapper();
+		if(e.getClick() == null) //For other types of clicks besides left, right, middle
+		{
+			e.setCanceled(true);
+			return;
+		}
+		else if(item.getItemStack() == null)
+		{
+			return;
+		}	
+		else if(e.getView() == InventoryView.BOTTOM)
+		{
+			if(e.getClick() == Click.SHIFT_LEFT || e.getClick() == Click.SHIFT_RIGHT)
+			{
+				if(!this.canStack(gui, e.getInventoryWrapper(), item))
```

Comment:
Codacy found an issue: [These nested if statements could be combined](https://app.codacy.com/manual/virustotalop/DynamicGui/pullRequest?prid=5724759)